src/html/parse-frontmatter.js

```diff
@@ -9,315 +9,27 @@
-
-    const txt = str.slice(fst.offEnd, !last ? undefined : last.offStart);
-    let data;
-    try {
-      data = yaml.safeLoad(txt);
-    } catch (e) {
-      return warn(e, `Exception ocurred while parsing yaml: ${e}`);
-    }
-
-    if (type(data) !== Object) {
-      return warn(null,
-        'Found ambigous frontmatter block: Block contains valid yaml, but '
-          + `it's data type is ${type(data)} instead of Object.`
-          + 'Make sure your yaml blocks contain only key-value pairs at the root level!');
-    }
-
-    return {
-      type: 'frontmatter',
-      payload: data,
-      start: fst.idx,
-      end: last.idx,
-    };
-  });
-
-  // Preprocessing
-  return pipe(
-    enumerate(mdast.children),
-    // Find any potential frontmatter starts/ends in the mdast
-    /* eslint-disable-next-line no-unused-vars */
-    filter(([idx, nod]) => isPotential(nod)),
-    // Filter out dom nodes based on their actual text content;
-    // this filters out HRs made from other characters or setext
-    // headings with more than three dashes...
-    //
-    // And: Perform some more sophisticated feature extraction on the nodes
-    map(([idx, nod]) => {
-      const mat = nodeStr(idx).match(re(`(?<=^|\\n)---${hspace}*\\n?$`));
-      if (!mat) {
-        return null;
-      }
-      // Offset of the actual separator line (this may deviate from the)
-      const offStart = mat.index + start(idx);
-      const offEnd = offStart + size(mat[0]);
-      // Is there a new line or EOF before/after the separator?
-      const before = Boolean(str.slice(0, offStart).match(re(`(^|(^|\\n)${hspace}*\\n)$`)));
-      const after = Boolean(str.slice(offEnd).match(re(`^(${hspace}*(\\n${hspace}*(\\n|$))|$)`)));
-      return {
-        idx, nod, offStart, offEnd, before, after,
-      };
-    }),
-    filter(identity),
-    // Pair up two fences each; we even do this if there is only a single
-    // fence (even though by definition that could never form a frontmatter
-    // block) in order to warn about ambiguous nodes
-    lookahead(1, null),
-    // Filter out pairs in which both the start and the end is definately
-    // a settext heading or <hr>
-    reject(([fst, last]) => toIgnore(fst) && toIgnore(last)),
-    // Decide which blocks to ignore, which deserve warnings and which
-    // are actual frontmatter
-    procwarnigns,
-    filter(identity),
-    // Filter out false positive warnings for pseudo frontmatter blocks
-    // before actual frontmatter (warning gets invalidated by the fact
-    // that it DIRECTLY PRECEDES an actual frontmatter block)
-    lookahead(1, null),
-    reject(([val, next]) => true
-      && val.type === 'warning'
-      && val.warning.startsWith('Found ambigous frontmatter')
-      && next
-      && next.type === 'frontmatter'
-      && val.end === next.start),
-    /* eslint-disable-next-line no-unused-vars */
-    map(([val, next]) => val),
-  );
-};
+/**
+ * @deprecated
+ */
+const findFrontmatter = () => [];
```

Comment:
Why an empty array? Wouldn't `undefined` work as well?

Reply:
AFAICS, the original `findFrontmatter()` returned an array.